src/hats/io/file_io/file_io.py

```diff
@@ -107,18 +107,21 @@ def load_csv_to_pandas(file_pointer: str | Path | UPath, **kwargs) -> pd.DataFra
 
 
 def load_csv_to_pandas_generator(
-    file_pointer: str | Path | UPath, chunksize=10_000, **kwargs
+    file_pointer: str | Path | UPath, *, chunksize=10_000, open_mode=None, compression=None, **kwargs
 ) -> Generator[pd.DataFrame]:
     """Load a csv file to a pandas dataframe
     Args:
         file_pointer: location of csv file to load
-        file_system: fsspec or pyarrow filesystem, default None
+        chunksize (int): number of rows to load per chunk
+        compression (str): for compressed CSVs, the manner of compression. e.g. 'gz', 'bzip'.
         **kwargs: arguments to pass to pandas `read_csv` loading method
     Returns:
         pandas dataframe loaded from CSV
     """
     file_pointer = get_upath(file_pointer)
-    with file_pointer.open("r", **kwargs) as csv_file:
+    if open_mode is None:
+        open_mode = "r" if compression is None else "rb"
```

Comment:
```suggestion
    open_mode = "rb" if open_mode is None else open_mode 
```

Is there any good reason to use "r" by default? I'd always use "rb" - think about non-UTF8 encodings, etc.

For example:
```python
from pathlib import Path
from tempfile import TemporaryDirectory
import pandas as pd

with TemporaryDirectory() as d:
    path = Path(d) / "koi8-r.csv"
    with path.open(encoding="koi8-r", mode="w") as fh:
        fh.write("col1,col2\nыыы,яяя\n")
    with path.open(mode="rb") as fh:
        df1 = pd.read_csv(fh, encoding="koi8-r")  # works
    with path.open(mode="r") as fh:
        df2 = pd.read_csv(fh)  # fails
```

Reply:
Added this as a test - thank you SO SO SO MUCH for providing this snippet!

As for the compression question, de-compression is happening in the call to open, and happening on the fsspec side. Pandas will not handle de-compression when provided a file handle, and we HAVE to provide a file handle, because pandas doesn't support passing a fsspec object or UPath.